@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -724,6 +731,56 @@ export class TileGeometryCreator {
                     object.userData.geometryId = srcGeometry.uuid;
                 }
 
+                if (isFillTechnique(technique)) {
+                    object.onBeforeRender = chainCallbacks(
+                        object.onBeforeRender,
+                        (_renderer, _scene, _camera, _geometry, _material) => {
+                            const clipping = getPropertyValue(
+                                technique.clipping,
+                                mapView.zoomLevel
+                            );
+                            const transparent = getPropertyValue(
+                                technique.transparent,
+                                mapView.zoomLevel
+                            );
+                            const clip =
+                                clipping || (clipping === undefined && transparent === true);
+                            if (!clip) {
+                                return;
+                            }
+                            if (_material.clippingPlanes === null) {
+                                _material.clippingPlanes = this.clippingPlanes;
+                                // TODO: Add clipping for Spherical projection.
+                            } else if (mapView.projection.type === ProjectionType.Planar) {
+                                const planes = _material.clippingPlanes;
```

Comment:
I wonder if we could do this only once per material in the tile and not for every object in the tile. Therefore we would need sth. like a Tile.onBeforeRender or Tile.updateTechniques callback. I remember this was planned anyways by @robertoraggi , not sure what's the state though.

Reply:
there's nothing in our code preventing our users to share a material with objects coming from different tiles, so unless we explicitly prevent sharing materials we need to have a per-object call for setting up the clipping planes.

Reply:
What do you mean? The materials are created by us and internal to the tile objects. We don't expose them. Of course a user can dig trough the objects of the tile and get access to the materials but that is such an advanced usecase where the user is most likely aware that the material needs to be cloned.
Maybe we can discuss this tomorrow face 2 face but this was my understanding how we will also do the interpolations in the future (i.e. Tile.updateTechniques will evaluate all interpolations and update the materials accordingly)

Reply:
All it's required by our current DataSource/Tile APIs is that a tile `Tile` provides a list of threejs objects for each `tile.objects`. For example, the DebugTIleDataSource defines only one material `debugMaterial` https://github.com/heremaps/harp.gl/blob/master/%40here/harp-debug-datasource/lib/DebugTileDataSource.ts#L19 and it uses it for each tile https://github.com/heremaps/harp.gl/blob/master/%40here/harp-debug-datasource/lib/DebugTileDataSource.ts#L71

many examples and apps using harp.gl do the same.

Reply:
@robertoraggi But that's a different usecase. These datasources don't use a Technique. This change is adding the clipping planes only to the materials that are created for the fill technique-

Reply:
I don't fully understand this conversation, but it sounds like it is referring to some refactoring we should do in the material, and their handling? And is orthogonal, though somewhat related to this patch?